dattri/func/projection.py

```diff
@@ -753,6 +755,18 @@ def get_eigenspace(self) -> None:
         # proj dim is the value of k
         self.eigvals, self.eigvecs = self._distill(appr_mat, proj, self.proj_dim)
 
+        # prevent from negative eigvals
+        if self.proj_dim > torch.sum(self.eigvals > 0):
```

Comment:
Maybe change "> 0" to "> self.tol**2" to improve numerical stability

Reply:
Sure